viewer/db.js

```diff
@@ -987,31 +1003,262 @@ exports.getHunt = async (id) => {
+
+  try {
+    // fetch the version of the remote shortcuts index to check if the local shortcuts index
+    // is up to date. if not, something has changed in the remote index and we need to sync
+    const version = await getShortcutsVersion();
+
+    if (version === internals.localShortcutsVersion) { return; } // version's match, stop!
+
+    console.log(msg);
+
+    // fetch shortcuts from remote and local indexes
+    const [{ body: remoteResults }, { body: localResults }] = await Promise.all([
+      exports.searchShortcuts({}), exports.searchShortcutsLocal({})
+    ]);
+
+    // compare the local shortcuts to the remote shortcuts to determine
+    // if any shortcuts have been deleted from the remote db
+    for (const localShortcut of localResults.hits.hits) {
+      let missing = true;
+      for (const remoteShortcut of remoteResults.hits.hits) {
+        if (remoteShortcut._id === localShortcut._id) {
+          missing = false; // we found it, it's not missing
+          break;
+        }
+      }
+      // if we get here without the missing flag set to false
+      if (missing) { // it's missing from the remote db
+        internals.client7.delete({ // remove the shortcut from the local db
+          index: internals.localShortcutsIndex,
+          id: localShortcut._id,
+          refresh: true
+        });
+      }
+    }
+
+    // compare remote shortcuts to local shortcuts to determine if any
+    // shortcuts have been added or updated from the remote db
+    for (const remoteShortcut of remoteResults.hits.hits) {
+      let missing = true;
+      for (const localShortcut of localResults.hits.hits) {
+        if (remoteShortcut._id === localShortcut._id) {
+          missing = false; // found it, check if we need to update it
+          if (remoteShortcut._version !== localShortcut._version) {
+            internals.client7.index({ // update the shortcut in the local db
+              id: remoteShortcut._id,
+              index: internals.localShortcutsIndex,
+              body: remoteShortcut._source,
+              version_type: 'external',
+              version: remoteShortcut._version
+            });
+          }
+        }
+      }
+      // if we get here without the missing flag set to false
+      if (missing) { // it's missing from the local db
+        internals.client7.index({ // add the shortcut in the local db
+          id: remoteShortcut._id,
+          index: internals.localShortcutsIndex,
+          body: remoteShortcut._source,
+          version_type: 'external',
+          version: remoteShortcut._version
+        });
+      }
+    }
+
+    internals.localShortcutsVersion = version;
+  } catch (err) {
+    console.log(`ERROR - ${msg}:`, err);
+  }
+}
 exports.searchShortcuts = async (query) => {
+  return internals.usersClient7.search({
+    index: internals.remoteShortcutsIndex, body: query, rest_total_hits_as_int: true, version: true
```

Comment:
need size:10000 here, otherwise default is 10 results

Reply:
I'm not sending this function the query with `size: 10000`. I didn't want to break the client. It sends a size for pagination.